### PR TITLE
Fix RPC response URI validator

### PR DESCRIPTION
**Pull request by evshary**
According to the spec, `validateRpcResponse(UUri): UStatus` shoud return false while `UUri.resource.name != "rpc" || UUri.resource.instance != "response" || UUri.resource.id != 0`
https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc#3-validators

### CODE REVIEW

File: src/uri/validator/urivalidator.rs [hunk 1]
```diff
@@ -163,9 +163,9 @@ impl UriValidator {
                     .as_ref()
                     .map_or(false, |instance| instance.contains("response"));
 
-                let has_non_zero_id = resource.id.map_or(false, |id| id != 0);
+                let has_zero_id = resource.id.map_or(false, |id| id == 0);
 
-                return has_valid_instance || has_non_zero_id;
+                return has_valid_instance && has_zero_id;
```
Comment (PLeVasseur) [inline src/uri/validator/urivalidator.rs:168]: Agreed, this is a good catch!

File: src/uri/validator/urivalidator.rs [hunk 2]
```diff
@@ -163,9 +163,9 @@ impl UriValidator {
                     .as_ref()
                     .map_or(false, |instance| instance.contains("response"));
```
Comment (PLeVasseur) [inline src/uri/validator/urivalidator.rs:164]: I guess technically the spec says it should be equal to, not that it should contain "response". Shouldn't we check for equality?

Tagging @AnotherDaniel in here to sanity check me.

Comment (PLeVasseur) [inline src/uri/validator/urivalidator.rs:164]: Also noticed that we're not checking for `name` to be "rpc", unlike within `is_rpc_method()`, even though the spec says to check the name is "rpc" for both validators.

Am I reading that right? Can you check me here again @AnotherDaniel?

Comment (AnotherDaniel) [reply src/uri/validator/urivalidator.rs:164]: You're correct. The 'contains' is a leftover from me dumbly doing a 1:1 convert from up-java, but the spec requires equality.

Comment (evshary) [reply src/uri/validator/urivalidator.rs:164]: Maybe I'm wrong, but I guess it's intended to check "rpc" by using `is_rpc_method` [here](https://github.com/eclipse-uprotocol/up-rust/pull/52/files#diff-50d5281e2de4d6b2de3ffb2294e4a0bb1fe84a474bd1595b080e740cba9e2aacR159).

Comment (evshary) [reply src/uri/validator/urivalidator.rs]: I've updated it.

Comment (PLeVasseur) [reply src/uri/validator/urivalidator.rs]: [Spec says](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc#3-validators) we should check for it:
image

Comment (evshary) [reply src/uri/validator/urivalidator.rs]: Updated.

**Review by AnotherDaniel — APPROVED** [review]
LGTM

**Review by PLeVasseur — APPROVED** [review]
LGTM

File: src/uri/validator/urivalidator.rs [hunk 3]
```diff
@@ -1145,11 +1146,7 @@ mod tests {
             name: "hartley".into(),
             ..Default::default()
         };
-        let resource = UResource {
-            name: "rpc".into(),
-            id: Some(19999),
-            ..Default::default()
-        };
+        let resource = UResourceBuilder::for_rpc_response();
```
Comment (sophokles73) [inline src/uri/validator/urivalidator.rs:1149]: can we please keep the code that explicitly specifies the (valid) resource field values?

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**